src/main/java/com/gtnewhorizons/angelica/transform/compat/transformers/specific/ImmersiveEngineeringTransformer.java

```diff
@@ -0,0 +1,109 @@
+        final byte[] bytes = cw.toByteArray();
+        AngelicaTweaker.LOGGER.info("[AngelicaCompat]Extra Transformers: Applied ImmersiveEngineeringTransformer");
+        AngelicaTweaker.dumpClass(transformedName, basicClass, bytes, this);
+        return bytes;
+    }
+
+    private void staticRenderPassPatcher(ClassNode cn) {
+        for (MethodNode mn : cn.methods) {
+            if (mn.name.equals("renderWorldBlock")) {
+                for (int i = 0; i < mn.instructions.size(); i++) {
+                    AbstractInsnNode ain = mn.instructions.get(i);
+                    // So far every one of these has been named renderPass, eventually might need to make the name dynamic and possibly care about owner?
+                    // Maybe this is common in other mods, possibly could generalize this transformation further if it comes up more.
+                    if (ain instanceof FieldInsnNode fin && fin.getOpcode() == Opcodes.GETSTATIC && fin.name.equals("renderPass")) {
+                        MethodInsnNode getRenderPass = new MethodInsnNode(Opcodes.INVOKESTATIC, "net/minecraftforge/client/ForgeHooksClient", "getWorldRenderPass", "()I");
+                        mn.instructions.insert(fin, getRenderPass);
+                        mn.instructions.remove(fin);
+                    }
+                }
+            }
+        }
+    }
+
+    private void transformClientUtils(ClassNode cn) {
+        for (MethodNode mn : cn.methods) {
+            // This patches an if (world != null) check to just be if (false) in order to make the block of code not run
+            // The block of code in question calls OpenGL to update the lightmap texture coordinates, which is illegal from an ISBRH
+            // in Angelica. It doesn't seem to make anything not work by causing this to not happen.
+            if (mn.name.equals("renderStaticWavefrontModelWithIcon") || mn.name.equals("renderStaticWavefrontModel")) {
+                if (
+                    mn.desc.equals("(Lnet/minecraft/world/IBlockAccess;IIILnet/minecraftforge/client/model/obj/WavefrontObject;Lnet/minecraft/util/IIcon;Lnet/minecraft/client/renderer/Tessellator;Lblusunrize/immersiveengineering/common/util/chickenbones/Matrix4;Lblusunrize/immersiveengineering/common/util/chickenbones/Matrix4;IZFFF[Ljava/lang/String;)V")
+                    || mn.desc.equals("(Lnet/minecraft/world/IBlockAccess;IIILnet/minecraftforge/client/model/obj/WavefrontObject;Lnet/minecraft/client/renderer/Tessellator;Lblusunrize/immersiveengineering/common/util/chickenbones/Matrix4;Lblusunrize/immersiveengineering/common/util/chickenbones/Matrix4;IZFFF[Ljava/lang/String;)V")
+                ) {
+                    System.out.println("We have arrived");
+                    for (int i = 0; i < mn.instructions.size(); i++) {
+                        AbstractInsnNode ain = mn.instructions.get(i);
+                        if (ain instanceof JumpInsnNode jump && jump.getOpcode() == Opcodes.IFNULL) {
```

Comment:
what happens to the object that's currently loaded on the stack ? have you tested this ? I bet it doesn't work

Reply:
I've tested it and I haven't seen any problem with it. The particular bytecode this is targeting, in either branch of the jump, goes directly into another `ALOAD` in a new frame. It at least doesn't seem to break anything and my testing has hit this particular codepath a ton